### ISSUE TITLE
Not able to create custom Deep-Link Logic without errors 

### ISSUE THREAD

**Issue by Raqhael** [supabase/supabase-flutter#433]
**Describe the bug**
The Deep-Linking for GoTrue works neatly and is easily configurable. Ive set a custom scheme and add it to the URL-Config and everything works fine for OAuth-Redicrecting.

The Problem is: I want to add custom Deep-Linking functionality. I want to e.g. have a link http://mydomain.com/somequery/... which should redirect to a specific Page. And i get it to somewhat work, but no matter what link i use, it will go through the Gotrue-Deeplink handler (`_handleDeeplink`). And when it does not find a `access_token` (in other words its not a deep link meant for Supabase-OAuth) it will throw an Error:

```
I/flutter (21576): ***** SupabaseDeepLinkingMixin startAuthObserver
I/flutter (21576): ***** SupabaseAuthState handleDeeplink http://mydomain.com/tempdev
I/flutter (21576): onReceivedAuthDeeplink uri: http://mydomain.com/tempdev
E/flutter (21576): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: AuthException(message: No access_token detected., statusCode: null)
E/flutter (21576):
E/flutter (21576): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: AuthException(message: No access_token detected., statusCode: null)
E/flutter (21576):
I/flutter (21576): No access_token detected.
```

**To Reproduce**
1. Follow the Supabase Instruction to create an Flutter Auth App with Supabase (https://supabase.com/docs/guides/getting-started/tutorials/with-flutter). And get it to work with OAuth/Deep-linking
2. Start The App in VSC with F5 (Debug-Mode)
3. Add a http/s scheme with a custom domain and try to open it
4. See throwing and interrupting Error

**Expected behavior**
Instead of a Error, output a warning (?). Or somehow add a way to handle custom Deeplinks. There is a comment stating `getInitialLink/getInitialUrl` should only be andled once in your apps lifetime (supabase_flutter/lib/src/supabase_auth.dart:221). Is this meant only for this handleInitial URL aswell? Or to any getInitialURL when building custom uni_link handler ("disconnected" from supabase, but in the same project).

**Version (please complete the following information):**

On Windows
├── supabase_flutter 1.0.1
│   ├── supabase 1.0.1
│   │   ├── functions_client 1.1.1
│   │   ├── gotrue 1.4.1
│   │   ├── postgrest 1.2.2
│   │   ├── realtime_client 1.0.3
│   │   └── storage_client 1.2.3

**Additional context**
-

**Comment by dshukertjr** [comment]
Hi @Raqhael,

This issue was fixed in v1.2.1 of supabase_flutter. Would you be able to upgrade your supabase_flutter version and try it out?

**Comment by Raqhael** [comment]
Hi @dshukertjr yes this fixed it for me! Thank you!
